ds4drv/__main__.py

```diff
@@ -108,26 +109,55 @@ def read_report(self):
     def run(self):
         self.loop.run()
 
-    def exit(self, *args):
+    def exit(self, *args, error = True):
         if self.device:
             self.cleanup_device()
 
-        self.logger.error(*args)
-        self.error = True
+        if error == True:
+            self.logger.error(*args)
+            self.error = True
+        else:
+            self.logger.info(*args)
 
 
 def create_controller_thread(index, controller_options, dynamic=False):
     controller = DS4Controller(index, controller_options, dynamic=dynamic)
 
     thread = Thread(target=controller.run)
-    thread.daemon = True
     thread.controller = controller
     thread.start()
 
     return thread
 
 
+class SigintHandler(object):
+
+    def __init__(self, threads):
+        self.threads = threads
+
+    def cleanup_controller_threads(self):
+        for thread in self.threads:
+            thread.controller.exit("Cleaning up...", error=False)
+            thread.controller.loop.stop()
+            thread.join()
+
+    def __call__(self, signum, frame):
+        old_handler = signal.getsignal(signum)
+        signal.signal(signum, signal.SIG_DFL)
+
+        self.cleanup_controller_threads()
+
+        signal.signal(signum, old_handler)
```

Comment:
I would maybe set it to default (to allow force quit with another ^C) and not restore the handler. It's practical and simple. But I don't have a strong opinion here, so you can decide which option is better.

Squash the commits when you think this is ready for merging.